src/pubsub/ua_pubsub_writergroup.c

```diff
@@ -903,15 +903,8 @@ generateNetworkMessage(UA_PubSubConnection *connection, UA_WriterGroup *wg,
 
     networkMessage->version = 1;
     networkMessage->networkMessageType = UA_NETWORKMESSAGE_DATASET;
-    if(connection->config->publisherIdType == UA_PUBSUB_PUBLISHERID_NUMERIC) {
-        networkMessage->publisherIdType = UA_PUBLISHERDATATYPE_UINT16;
-        networkMessage->publisherId.publisherIdUInt32 =
-            connection->config->publisherId.numeric;
-    } else if(connection->config->publisherIdType == UA_PUBSUB_PUBLISHERID_STRING) {
-        networkMessage->publisherIdType = UA_PUBLISHERDATATYPE_STRING;
-        networkMessage->publisherId.publisherIdString =
-            connection->config->publisherId.string;
-    }
+    networkMessage->publisherIdType = connection->config->publisherIdType;
```

Comment:
Please add a comment that this is a shallow copy of the string.
And that the config needs to be stable until the message is sent.
And that we must not "clean up" the PublisherId in network message after sending.

Also add such a comment to the header file where the networkmessage is defined.